pkg/getty/etc/profile.d/namespace.sh

```diff
@@ -1 +1,23 @@
-export PS1="(ns: getty) $PS1"
+#/bin/sh
+
+in_a_ns() {
+	IN_A_NS=0
+	NOT_IN_A_NS=1
+
+	# If PID 1 != /sbin/init then we are in a container
+	if ! pgrep /sbin/init > /dev/null; then
+		return $IN_A_NS
+	fi
+
+	# If containerd or runc are in the pstree then we're probably in a namespace
+	if pstree | grep -qE "(containerd|runc)"; then
+		return $IN_A_NS
+	fi
```

Comment:
@ijc in my experimentation, when you have an `init` getty, you don't see `containerd` or `runc`. You only see these when you are a `service` and sharing a `pid` namespace. Changing the check to look for the `mnt` namespace is probably the better one. I'll adjust appropriately.